Description: <iftop dosplays bar in colors>
 <long description that can span multiple lines, optional>
Author: <srix codeporukki@gmail.com>
Origin: <upstream|backport|vendor|other>, <URL, required except if Author is present>
Bug: <URL to the upstream bug report if any, implies patch has been forwarded, optional>
Bug-<Vendor>: <URL to the vendor bug report if any, optional>
Forwarded: <URL|no|not-needed, useless if you have a Bug field, optional>
Applied-Upstream: <version|URL|commit, identifies patches merged upstream, optional>
Reviewed-by: <name and email of a reviewer, optional>
Last-Update: <2014-05-24, last update of the meta-information, optional>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/ui.c
+++ b/ui.c
@@ -91,6 +91,11 @@ time_t helptimer = 0;
 char helpmsg[HELP_MSG_SIZE];
 int dontshowdisplay = 0;
 
+#define COLOR_PAIR_SENT 1
+#define COLOR_PAIR_RECV 2
+#define COLOR_PAIR_BOTH 3
+
+
 /*
  * Compare two screen lines based on bandwidth.  Start comparing from the 
  * specified column
@@ -346,12 +351,13 @@ void draw_line_total(float sent, float r
     }
 }
 
-void draw_bar(float n, int y) {
+void draw_bar(float n, int y, short colorpair) {
     int L;
+    colorpair = has_colors()==TRUE?colorpair:0; /* set 0 if terminal is not color capable*/
     mvchgat(y, 0, -1, A_NORMAL, 0, NULL);
     L = get_bar_length(8 * n);
     if (L > 0)
-        mvchgat(y, 0, L + 1, A_REVERSE, 0, NULL);
+        mvchgat(y, 0, L + 1, A_REVERSE, colorpair, NULL);
 }
 
 void draw_line_totals(int y, host_pair_line* line, option_linedisplay_t linedisplay) {
@@ -366,17 +372,17 @@ void draw_line_totals(int y, host_pair_l
     if(options.showbars) {
       switch(linedisplay) {
         case OPTION_LINEDISPLAY_TWO_LINE:
-          draw_bar(line->sent[options.bar_interval],y);
-          draw_bar(line->recv[options.bar_interval],y+1);
+          draw_bar(line->sent[options.bar_interval],y, COLOR_PAIR_SENT);
+          draw_bar(line->recv[options.bar_interval],y+1,COLOR_PAIR_RECV);
           break;
         case OPTION_LINEDISPLAY_ONE_LINE_SENT:
-          draw_bar(line->sent[options.bar_interval],y);
+          draw_bar(line->sent[options.bar_interval],y,COLOR_PAIR_SENT);
           break;
         case OPTION_LINEDISPLAY_ONE_LINE_RECV:
-          draw_bar(line->recv[options.bar_interval],y);
+          draw_bar(line->recv[options.bar_interval],y,COLOR_PAIR_RECV);
           break;
         case OPTION_LINEDISPLAY_ONE_LINE_BOTH:
-          draw_bar(line->recv[options.bar_interval] + line->sent[options.bar_interval],y);
+          draw_bar(line->recv[options.bar_interval] + line->sent[options.bar_interval],y,COLOR_PAIR_BOTH);
           break;
       }
     }
@@ -787,6 +793,14 @@ void ui_tick(int print) {
 
 void ui_curses_init() {
     (void) initscr();      /* initialize the curses library */
+    if( has_colors()==TRUE)
+    {
+        start_color();          /* Start color          */
+        use_default_colors();   /* retain terminal fg , bg colors */
+        init_pair(COLOR_PAIR_RECV, COLOR_GREEN,  -1);  /* Download color */
+        init_pair(COLOR_PAIR_SENT, COLOR_BLUE,   -1);
+        init_pair(COLOR_PAIR_BOTH, COLOR_MAGENTA,-1);
+    }
     keypad(stdscr, TRUE);  /* enable keyboard mapping */
     (void) nonl();         /* tell curses not to do NL->CR/NL on output */
     (void) cbreak();       /* take input chars one at a time, no wait for \n */
